WordPress/Classes/Extensions/URL+Helpers.swift

```diff
@@ -206,3 +206,16 @@ extension URL {
         return newComponents.url ?? self
     }
 }
+
+extension URL {
+    /// Appends query items to the URL.
+    /// - Parameter newQueryItems: The new query items to add to the URL. These will **not** overwrite any existing items but are appended to the existing list.
+    /// - Returns: The URL with added query items.
+    func appendingQueryItems(_ newQueryItems: [URLQueryItem]) -> URL {
+        var components = URLComponents(url: self, resolvingAgainstBaseURL: false)
+        var queryItems = components?.queryItems ?? []
+        queryItems.append(contentsOf: newQueryItems)
```

Comment:
I'm waffling a bit here.  Since this is a general extension I'm thinking about other uses cases. Particularly what is the best behavior for when a parameter already exists.  Having the parameter twice could produce unexpected behavior, especially if the values are different.  On the other hand it doesn't make sense to over optimize when there's no immediate need.
Thoughts?

Reply:
I think this is definitely a good point for the true/false values. I'm not quite sure about other parameters since some implementations will combine duplicates into an array (although, I think PHP takes the last value).

I'd like to submit a follow up with unit tests for this and could add some details and a possible change there. Adding the deduplication could probably prevent some unintended side effects.

Do you think that makes sense?

Reply:
Sounds good to me!